rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -2379,6 +2355,16 @@ public Coin getLockingCap() {
         return this.provider.getLockingCap();
     }
 
+    /**
+     * Returns the redeemScript of the current active federation
+     * @return Returns the redeemScript of the current active federation
+     */
+    public Script getActivePowpegRedeemScript() {
+        return activations.isActive(ConsensusRule.RSKIP293) ?
+                getActiveFederation().getRedeemScript() :
+                new Script(new byte[]{});
```

Comment:
I wonder if it would make more sense to return an Optional instead of an empty Script.
The one receiving the result of this method wouldn't know how to differentiate a real Powpeg redeemscript from a default empty one...

Reply:
I agree with you, I prefer use optional instead of creating a default empty script, as you already said can be misleading to the one consuming the method. In the case it returns an optional, what should return the bridge method that calls it? Can it return the same optional? Or should it return an empty byte array?

I mean this method:

![image](https://user-images.githubusercontent.com/94651730/152364855-e85931a4-4957-454d-9313-23adeee4799c.png)